File: Trabajo Practico 9 Vega_Gonzalo/backend/src/main/java/ar/utn/lab4/tp3/model/Instrumento.java
package ar.utn.lab4.tp3.model;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class Instrumento {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    @Builder.Default
    @Column(columnDefinition = "boolean default true")
    boolean alta = true;
    String instrumento;
    String marca;
    String modelo;
    String imagen;
    String precio;
    @Column(name = "costo_envio")
    String costoEnvio;
    @Column(name = "cantidad_vendida")
    Integer cantidadVendida;
    @Column(length = 500)
    String descripcion;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "categoria_instrumento")
    CategoriaInstrumento categoriaInstrumento;

    
}

File: Trabajo Practico 9 Vega_Gonzalo/frontend/src/entities/Instrumento.ts
import CategoriaInstrumento from "./CategoriaInstrumento";

export default interface Instrumento {
    id: string;
    alta : boolean
    instrumento: string;
    marca: string;
    modelo: string;
    imagen: string;
    precio: string;
    costoEnvio: string;
    cantidadVendida: string;
    descripcion: string;
    categoriaInstrumento : CategoriaInstrumento
  }
  